easy_transformer/EasyTransformer.py

```diff
@@ -94,21 +68,27 @@ def __init__(
             self.cfg.d_vocab = max(self.tokenizer.vocab.values()) + 1
             self.cfg.d_vocab_out = self.cfg.d_vocab
 
+    def __init_config__(self, config: Union[EasyTransformerConfig, Dict]):
+        if isinstance(config, Dict):
+            config = EasyTransformerConfig(**config)
+        elif isinstance(config, str):
+            raise ValueError(
+                "Please pass in a config dictionary or EasyTransformerConfig object. If you want to load a pretrained model, use EasyTransformer.from_pretrained() instead."
+            )
+        self.cfg = config
+
+    def __init_embeddings__(self):
         self.embed = Embed(self.cfg)
         self.hook_embed = HookPoint()  # [batch, pos, d_model]
 
         if self.cfg.positional_embedding_type != "rotary":
             self.pos_embed = PosEmbed(self.cfg)
             self.hook_pos_embed = HookPoint()  # [batch, pos, d__dictmodel]
 
-        self.blocks = nn.ModuleList(
-            [
-                TransformerBlock(self.cfg, block_index)
-                for block_index in range(self.cfg.n_layers)
-            ]
-        )
-
-        if self.cfg.normalization_type == "LN":
+    def __init__normalization__(self):
```

Comment:
Also overkill

Reply:
At least, called it final normalization or pre-unembed normalization

Reply:
paused on https://github.com/neelnanda-io/Easy-Transformer/pull/71#pullrequestreview-1181263509